docs/API/V1/README.md

```diff
@@ -8,6 +8,57 @@ Once you know the hostname of a Pbench Server, you can ask for the API
 configuration using the [endpoints](endpoints.md) API. This will report the
 server's version and a list of all API end points supported by the server.
 
+## Pbench Server resources
+
+### Datasets
+
+The dataset resource encompasses everything the Pbench Server knows about a
+performance run captured by the Pbench Agent through a "dataset tarball". This
+includes the physical files transferred from the agent along with all backend
+databases maintained by the server.
+
+The API identifies a dataset using the dataset resource ID, which is a hex
+representing a checksum hash of the dataset tarball. This ID can't change during
+the life of the dataset.
+
+When a dataset is deleted, all data maintained by the server associated with that
+dataset resource ID is deleted, including backend database records, unpacked file
+artifacts, and the archived tarball. The resource ID becomes invalid subsequently
+unless a new dataset is created with the same checksum value. (This is highly
+unlikely, unless the same Pbench Agent tarball is uploaded again.)
+
+### Users
+
+User resources are identified by a `username` property, which must be unique
+among all users registered on the Pbench Server. The user resource has a
+"user profile" that includes first and last name, and a contact email.
+
+A user resource is the "owner" of each [dataset](#datasets) managed by the
+Pbench Server. If a user is deleted, then any datasets owned by that user
+become orphaned; datasets with PUBLIC access are still accessible to other
```

Comment:
Do we cover somewhere what it means to be "orphaned"?

It might be good to reference another section here that talks about the behaviors of orphaned datasets.  An administrator of the Pbench Server would need to know what the impact will be on the server operation.

Reply:
I actually realized as I wrote this section that I don't believe I wrote a unit test to *verify* what happens when you delete a user. We probably need an admin mode API to change ownership of a dataset, and I wonder if maybe we should just refuse to delete a user if the user owns datasets rather than dealing with all this. That is, if you own datasets, then they need to be re-assigned *before* deleting the user.